File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/SignInViewModelOutputs.java
package com.travlog.android.apps.viewmodels.outputs;

import com.travlog.android.apps.libs.rx.Optional;

import io.reactivex.Observable;

public interface SignInViewModelOutputs {

    /**
     * Emits a boolean to determine whether or not the login button should be enabled.
     */

    Observable<Optional> signInSuccess();

    Observable<String> showFacebookAuthorizationErrorDialog();

    Observable<Boolean> setSignInButtonEnabled();
}

File: app/src/main/java/com/travlog/android/apps/viewmodels/outputs/SignInViewModelOutputs.kt
package com.travlog.android.apps.viewmodels.outputs

import io.reactivex.Completable

import io.reactivex.Observable

interface SignInViewModelOutputs {

    /**
     * Emits a boolean to determine whether or not the login button should be enabled.
     */

    fun showFacebookAuthorizationErrorDialog(): Observable<String>

    fun setSignInButtonEnabled(): Observable<Boolean>

    fun startMainActivity(): Completable
}
